language/tools/test_generation/src/transitions.rs

```diff
@@ -102,6 +106,115 @@ pub fn stack_local_polymorphic_eq(state: &AbstractState, index1: usize, index2:
     false
 }
 
+/// Determine whether the struct at the given index can be constructed from the values on
+/// the stack.
+pub fn stack_satisfies_struct_signature(
+    state: &AbstractState,
+    struct_index: StructDefinitionIndex,
+) -> bool {
+    let struct_def = state.module.struct_def_at(struct_index);
+    let struct_def = StructDefinitionView::new(&state.module, struct_def);
+    let field_token_views: Vec<SignatureTokenView<'_, CompiledModule>> = struct_def
+        .fields()
+        .into_iter()
+        .flatten()
+        .map(|field| field.type_signature().token())
+        .collect();
```

Comment:
Why collect this into a vec? You just iterate it later.

Reply:
Later on in the function body I use it to initialize a `SignatureToken::Struct` which expects a vector of tokens.

Reply:
Will be updated in next PR.